gfx_imgui.h

```diff
@@ -37,7 +37,7 @@ SOFTWARE.
 //! ImGui initialization/termination.
 //!
 
-GfxResult gfxImGuiInitialize(GfxContext gfx, char const **font_filenames = nullptr, uint32_t font_count = 0,
+GfxResult gfxImGuiInitialize(GfxContext const &gfx, char const **font_filenames = nullptr, uint32_t font_count = 0,
```

Comment:
I think I'd rather keep these public API functions without the `const &` thing.
Not massively important I suppose, but equally that copy of `GfxContext` isn't realistically likely to cause any performance issue 🙂

Reply:
The warnings are all about the size of the structs being larger than some threshold.
For reference:
sizeof(GfxContext) 80B
sizeof(GfxBuffer) 96B
sizeof(GfxTexture) 120B
sizeof(GfxAccelerationStructure) 80B

which is not huge but not trivially small either. The big size influence is all of these have a static char array as a member variable (fixed size at 65 chars) that has to be copied (which as its fixed size is potentially more costly to copy than strings that contain fewer chars). So you can see why analyzers would suggest they not be copied by value.